edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/IMetadataStore.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.Core
+{
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public interface IMetadataStore
+    {
+        Task SetModelId(string id, string modelId);
+
+        Task<Option<string>> GetModelId(string id);
+
+        Task SetProductInfo(string id, string productInfo);
```

Comment:
Do we still need these methods? Why not just GetMetadata/SetMetadata?

Reply:
I believe we still need SetProductInfo/SetModelId since we can encounter a situation where we have just product info or just modelId. But GetModelId and GetProductInfo could just become GetMetadata